src/search_dragon/external_apis/__init__.py

```diff
@@ -1,9 +1,61 @@
+import os
+import requests
+from search_dragon import logger
+
 class OntologyAPI:
     def __init__(self, base_url, api_id, api_name):
         self.base_url = base_url
         self.api_id = api_id
         self.api_name = api_name
 
+    def get_api_key(self, api_id):
+        if api_id == "umls":
```

Comment:
You could override this function at each API child class with an appropriate response. If the default response is the exception, you could throw that in the base class and only override the function in the child classes where it makes sense.